python/lsst/ctrl/mpexec/singleQuantumExecutor.py

```diff
@@ -233,3 +238,35 @@ def runQuantum(self, task, quantum, taskDef, butler):
                     f" it could happen due to inconsistent options between Quantum generation"
                     f" and execution") from exc
             butlerQC.put(task.metadata, ref[0])
+
+    def initGlobals(self, quantum, butler):
+        """Initialize global state needed for task execution.
+
+        Parameters
+        ----------
+        quantum : `~lsst.daf.butler.Quantum`
+            Single Quantum instance.
+        butler : `~lsst.daf.butler.Butler`
+            Data butler.
+
+        Notes
+        -----
+        There is an issue with initializing filters singleton which is done
+        by instrument, to avoid requiring tasks to do it in runQuantum()
+        we do it here when any dataId has an instrument dimension. Also for
+        now we only allow single instrument, verify that all instrument
+        names in all dataIds are identical.
+
+        This will need revision when filter singleton disappears.
+        """
+        oneInstrument = None
+        for datasetRefs in chain(quantum.predictedInputs.values(), quantum.outputs.values()):
+            for datasetRef in datasetRefs:
+                dataId = datasetRef.dataId
+                instrument = dataId.get("instrument")
+                if instrument is not None:
+                    if oneInstrument is not None:
+                        assert instrument == oneInstrument
```

Comment:
Can you add a message to this along the lines of "Currently require that only one instrument is used per graph" or something?